rootston/layer_shell.c

```diff
@@ -297,35 +297,32 @@ static void popup_handle_map(struct wl_listener *listener, void *data) {
 	struct roots_layer_surface *layer = popup->parent;
 	struct wlr_output *wlr_output = layer->layer_surface->output;
 	struct roots_output *output = wlr_output->data;
-	struct wlr_box geom;
-	memcpy(&geom, &popup->wlr_popup->geometry, sizeof(struct wlr_box));
-	geom.x += layer->geo.x;
-	geom.y += layer->geo.y;
-	wlr_output_damage_add_box(output->damage, &geom);
+	int ox = popup->wlr_popup->geometry.x;
```

Comment:
Can you sum `layer->geo.x` in the `ox` definition? Otherwise it's not really the position in output-local coords.